test/e2e/case3_mutation_recovery_test.go

```diff
@@ -136,11 +136,13 @@ var _ = Describe("Test unexpected policy mutation", func() {
 		rootPlc := utils.GetWithTimeout(
 			clientHubDynamic, gvrPolicy, case3PolicyName, testNamespace, true, defaultTimeoutSeconds,
 		)
-		Eventually(func() interface{} {
+		Eventually(func(g Gomega) interface{} {
 			plc = utils.GetWithTimeout(
 				clientHubDynamic, gvrPolicy, testNamespace+"."+case3PolicyName, "managed2", true, defaultTimeoutSeconds,
 			)
 
+			utils.RemovePolicyTemplateDBAnnotations(g, plc)
+
 			return plc.Object["spec"]
 		}, defaultTimeoutSeconds, 1).Should(utils.SemanticEqual(rootPlc.Object["spec"]))
```

Comment:
(Optional) This seems so common it could be part of a new `util` function that wraps both functions and returns the spec, something like:
```golang
		Eventually(
			utils.GetCleanPolicy(
				clientHubDynamic, 
				gvrPolicy,
				testNamespace+"."+case3PolicyName,
				"managed2", 
				true, 
				defaultTimeoutSeconds,
			), defaultTimeoutSeconds, 1).Should(utils.SemanticEqual(rootPlc.Object["spec"]))
```
With that, you can return spec and the error and Ginkgo will fail the test accordingly on error: https://onsi.github.io/gomega/#category-2-making-codeeventuallycode-assertions-on-functions

Reply:
I decided to skip this suggestion since time is tight.